diff --git a/files/dockerd.init b/files/dockerd.init
--- a/files/dockerd.init
+++ b/files/dockerd.init
@@ -44,6 +44,32 @@ boot() {
 	rc_procd start_service
 }
 
+uciadd_forward() {
+	# Variables
+	src_zone="docker"
+	dest_zone="wan"
+
+	exists=0
+
+	# Loop over all forwarding entries
+	for i in $(seq 0 $(($(uci show firewall | grep -c '^firewall.@forwarding') - 1))); do
+		if [ "$(uci get firewall.@forwarding[$i].src 2>/dev/null)" = "$src_zone" ] &&
+			[ "$(uci get firewall.@forwarding[$i].dest 2>/dev/null)" = "$dest_zone" ]; then
+			exists=1
+			break
+		fi
+	done
+
+	if [ $exists -eq 0 ]; then
+		logger -t "dockerd-init" -p notice "Adding firewall forwarding from '${src_zone}' to '${dest_zone}'"
+		uci add firewall forwarding
+		uci set firewall.@forwarding[-1].src="${src_zone}"
+		uci set firewall.@forwarding[-1].dest="${dest_zone}"
+		uci commit firewall
+	else
+		logger -t "dockerd-init" -p notice "Firewall forwarding from '${src_zone}' to '${dest_zone}' already exists"
+	fi
+}
 uciadd() {
 	local iface="${1}"
 	local device="${2}"
@@ -101,6 +127,8 @@ uciadd() {
 		uci_quiet commit firewall
 	fi
 
+	uciadd_forward
+
 	reload_config
 }
 
@@ -150,7 +178,7 @@ ucidel() {
 }
 
 process_config() {
-	local alt_config_file data_root log_level iptables ip6tables bip
+	local alt_config_file data_root log_level bip
 
 	[ -f /etc/config/dockerd ] || {
 		# Use the daemon default configuration
@@ -171,8 +199,6 @@ process_config() {
 
 	config_get data_root globals data_root "/opt/docker/"
 	config_get log_level globals log_level "warn"
-	config_get_bool iptables globals iptables "1"
-	config_get_bool ip6tables globals ip6tables "0"
 
 	# Don't add these options by default
 	# omission == docker defaults
@@ -195,8 +221,7 @@ process_config() {
 	json_init
 	json_add_string "data-root" "${data_root}"
 	json_add_string "log-level" "${log_level}"
-	json_add_boolean "iptables" "${iptables}"
-	json_add_boolean "ip6tables" "${ip6tables}"
+	json_add_string "firewall-backend" "nftables"
 	[ -z "${log_driver}" ] || json_add_string "log-driver" "${log_driver}"
 	[ -z "${bip}" ] || json_add_string "bip" "${bip}"
 	[ -z "${registry_mirrors}" ] || json_add_array "registry-mirrors"
@@ -221,8 +246,6 @@ process_config() {
 	fi
 	[ -z "${storage_driver}" ] || json_add_string "storage-driver" "${storage_driver}"
 	json_dump > "${DOCKERD_CONF}"
-
-	[ "${iptables}" -eq "1" ] && config_foreach iptables_add_blocking_rule firewall
 }
 
 start_service() {
@@ -250,49 +273,6 @@ service_triggers() {
 	procd_add_reload_trigger 'dockerd'
 }
 
-iptables_add_blocking_rule() {
-	local cfg="${1}"
-
-	local device=""
-	local extra_iptables_args=""
-
-	handle_iptables_rule() {
-		local interface="${1}"
-		local outbound="${2}"
-		local extra_iptables_args="${3}"
-
-		local inbound=""
-
-		. /lib/functions/network.sh
-		network_get_physdev inbound "${interface}"
-
-		[ -z "${inbound}" ] && {
-			logger -t "dockerd-init" -p notice "Unable to get physical device for interface ${interface}"
-			return
-		}
-
-		# Wait for a maximum of 10 second per command, retrying every millisecond
-		local iptables_wait_args="--wait 10 --wait-interval 1000"
-
-		# Ignore errors as it might already be present
-		iptables ${iptables_wait_args} --table filter --new DOCKER-USER 2>/dev/null
-		if ! iptables ${iptables_wait_args} --table filter --check DOCKER-USER --in-interface "${inbound}" --out-interface "${outbound}" ${extra_iptables_args} --jump REJECT 2>/dev/null; then
-			logger -t "dockerd-init" -p notice "Drop traffic from ${inbound} to ${outbound}"
-			iptables ${iptables_wait_args} --table filter --insert DOCKER-USER --in-interface "${inbound}" --out-interface "${outbound}" ${extra_iptables_args} --jump REJECT
-		fi
-	}
-
-	config_get device "${cfg}" device
-
-	[ -z "${device}" ] && {
-		logger -t "dockerd-init" -p notice "No device configured for ${cfg}"
-		return
-	}
-
-	config_get extra_iptables_args "${cfg}" extra_iptables_args
-	config_list_foreach "${cfg}" blocked_interfaces handle_iptables_rule "${device}" "${extra_iptables_args}"
-}
-
 stop_service() {
 	if /etc/init.d/dockerd running; then
 		service_stop "/usr/bin/dockerd"